File: Difficulty/Easy/[118] Pascal's Triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0: return []
        if numRows == 1: return [[1]]
        if numRows == 2: return [[1], [1, 1]]
        res = [[1], [1, 1]]
        for i in range(3, numRows+1):
            tmp = [0] * i
            for j in range(1, i + 1):
                if j == 1 or j == i:
                    tmp[j - 1] = 1
                else:
                    tmp[j - 1] = res[i - 2][j - 1] + res[i - 2][j - 2]
            res.append(tmp)
        return res

File: Difficulty/Easy/[118] Pascal's Triangle/solution.cpp
/*
 * @lc app=leetcode id=118 lang=cpp
 *
 * [118] Pascal's Triangle
 */

// @lc code=start
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if (numRows == 0) return vector<vector<int>>();
        if (numRows == 1) return vector<vector<int>>(1, vector<int>(1, 1));
        if (numRows == 2) return vector<vector<int>>{{1}, {1, 1}};
        vector<vector<int>> res{{1}, {1, 1}};
        for (int i = 3; i <= numRows; ++i) {
            vector<int> tmp(i, 0);
            for (int j = 1; j <= i; ++j){
                if (j == 1 || j == i)
                    tmp[j-1] = 1;
                else
                    tmp[j-1] = res[i - 2][j - 1] + res[i - 2][j - 2];
            }
            res.push_back(tmp);
        }
        return res;
    }
};
// @lc code=end

